### PR TITLE
library: Add TabView entry

### LINKED ISSUE #291: AdwTabView
https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/class.TabView.html

Let's keep it simple and include

* TabView
* TabOverview
* TabBar

in the same window

checkout the libadwaita Tab View demo but let's keep it simpler

### PR DESCRIPTION
**Pull request by AkshayWarrier**
Closes #291 
Adds demo entry for ``AdwTabView``. Also creates a new section "Navigation" in the library.

### CODE REVIEW

**Review by andyholmes — APPROVED** [review]
I think this looks really great, what do you think @sonnyp?

File: src/Library/demos/View Switcher/main.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "View Switcher",
-  "category": "user_interface",
+  "category": "navigation",
```
[the remark this answers is not in the document]
Comment (sonnyp) [reply src/Library/demos/View Switcher/main.json:3]: @andyholmes keep in mind that the merge strategy is Squash so all commits get squashed into a single commit that has the PR title as commit message.

[the remark this answers is not in the document]
Comment (andyholmes) [reply src/Library/demos/View Switcher/main.json:3]: Okay, as long as your fine with those changes being in one commit :)

[the remark this answers is not in the document]
Comment (sonnyp) [reply src/Library/demos/View Switcher/main.json:3]: I don't mind

**Comment by sonnyp** [comment]
Looks fantastic. 

One small problem is it doesn't support dragging the tab out of the window.

> Adwaita-Critical: AdwTabView::create-window handler must not return NULL

I think it's fine, supporting it would make the demo convoluted

**Pull request merged**